src/components/ApplicationDetailsModal.tsx

```diff
@@ -39,7 +51,24 @@ function ApplicationDetailsModal({ row }: { row: ApplicationDetails }) {
         </div>
         <div>
           <h1 className="text-base font-bold">Status:</h1>
-          <p className="pb-2 text-base">{row?.statusKey}</p>
+          <p className="pb-2 text-base">
+            {isProposalOpen ? row?.statusKey : 'Please Refresh Page'}
+            {isProposalOpen && row?.statusKey === 'OPEN' ? (
+              <Button
+                onClick={async () => {
+                  setIsProposalOpen(false)
+                  await acceptApplication.mutateAsync({
+                    proposalId: row.proposalId,
+                    proposalApplicationId: row.id,
+                    applicantEmail: row.email,
+                  })
+                  toast.success('Application accepted successfully!')
```

Comment:
refresh the page automatically here? the flow currently finishes in 500ms. we should get the response here, right? then instead or in addition to the toast.success, we can refetch the data automatically

Reply:
@rschlaefli 
What if the internet connection is poor?